### PR TITLE
Feature Change Keyword and Recipient Name searches to use TS-Vector columns

### PR DESCRIPTION
**Pull request by tony-sappe**
**High level description:**
Utilize a datatype which provides _very performant_ text search capabilities. This will improve both the keyword and recipient name filters in Advanced Search.

**Technical details:**
- Using the Postgres to_tsvector function to create ts_vector columns which are designed for very fast full text search queries. 
- Use the Django Postgres contrib module's `SearchVectorField` which supports correct postgres syntax for ts_vector queries
- Tweaked some existing indexes in matviews to not be compound since there was no benefits. (action_date + pulled_from => pulled_from)

**Link to JIRA Ticket:**
No ticket, High-level priority after GoLive

**The following are ALL required for the PR to be merged:**
- [x] Backend review completed
- [ ] Matview impact assessment completed
- [X] Frontend impact assessment completed N/A
- [X] Data validation completed N/A
- [x] API Performance evaluation completed (See future comments)

**Notes**
To not break existing code, existing columns were kept in the materialized views. Once these code changes propagate up to production, the deprecated columns `award_id_string` and `keyword_string` can be removed from the matview SQL.

### CODE REVIEW

**Comment by tony-sappe** [comment]
#### Performance Delta 1 
**`/api/v2/search/spending_by_award_count/`**
```{"filters":{"keyword":"e-corp","time_period":[{"start_date":"2017-10-01","end_date":"2018-09-30"},{"start_date":"2016-10-01","end_date":"2017-09-30"}]}```

Original: Timeout
New: 4s

**Comment by tony-sappe** [comment]
#### Performance Delta 2
**`/api/v2/search/spending_by_award/`**
```{"filters":{"time_period":[{"start_date":"2017-10-01","end_date":"2018-09-30"},{"start_date":"2016-10-01","end_date":"2017-09-30"},{"start_date":"2014-10-01","end_date":"2015-09-30"},{"start_date":"2015-10-01","end_date":"2016-09-30"}],"keyword":"solar", "award_type_codes":["A","B","C","D"]},"fields":["Award ID","Recipient Name","Start Date","End Date","Award Amount","Awarding Agency","Awarding Sub Agency","Contract Award Type"],"page":1,"limit":60,"sort":"Award Amount","order":"desc"}```

Original: 3s
New: 1.5s

**Comment by tony-sappe** [comment]
#### Performance Delta 3
**`/api/v2/search/spending_over_time/`***
```{"group":"fiscal_year","filters":{"keyword":"lockheed","time_period":[{"start_date":"2017-10-01","end_date":"2018-09-30"},{"start_date":"2016-10-01","end_date":"2017-09-30"}]}}```

Original: Timeout
New: 3.4s

**Comment by tony-sappe** [comment]
#### Performance Delta 4

**`/api/v2/search/spending_over_time/`**
```{"group":"fiscal_year","filters":{"time_period":[{"start_date":"2017-10-01","end_date":"2018-09-30"},{"start_date":"2016-10-01","end_date":"2017-09-30"}],"recipient_search_text":["Booz Allen"]}}```

Original: 48s
New: 5s

File: usaspending_api/database_scripts/matview_generator/summary_view_cfda_number.json [hunk 1]
```diff
@@ -40,9 +40,12 @@
     }, {
       "name": "type",
       "columns": [
-        {"name": "\"action_date\"", "order": "DESC NULLS LAST"},
         {"name": "\"type\""}
       ]
+    }, {
```
Comment (dpb-bah) [inline usaspending_api/database_scripts/matview_generator/summary_view_cfda_number.json]: Can we add these added `pulled_from` and removed `action_date` checks to the technical notes?

**Pull request merged**